src/components/common/CheckWallet/index.test.tsx

```diff
@@ -110,6 +133,63 @@ describe('CheckWallet', () => {
     expect(allowContainer.querySelector('button')).not.toBeDisabled()
   })
 
+  it('should not disable the button for delegates', () => {
+    ;(useIsSafeOwner as jest.MockedFunction<typeof useIsSafeOwner>).mockReturnValueOnce(false)
+    ;(useIsWalletDelegate as jest.MockedFunction<typeof useIsWalletDelegate>).mockReturnValueOnce(true)
+
+    const { container } = renderButton()
+
+    expect(container.querySelector('button')).not.toBeDisabled()
+  })
+
+  it('should disable the button for counterfactual Safes', () => {
+    ;(useIsSafeOwner as jest.MockedFunction<typeof useIsSafeOwner>).mockReturnValueOnce(true)
+
+    const safeAddress = faker.finance.ethereumAddress()
+    const mockSafeInfo = {
+      safeAddress,
+      safe: extendedSafeInfoBuilder()
+        .with({ address: { value: safeAddress } })
+        .with({ deployed: false })
+        .build(),
+    }
+
+    ;(useSafeInfo as jest.MockedFunction<typeof useSafeInfo>).mockReturnValueOnce(
+      mockSafeInfo as unknown as ReturnType<typeof useSafeInfo>,
+    )
+
+    const { container } = renderButton()
+
+    expect(container.querySelector('button')).toBeDisabled()
+    expect(container.querySelector('span[aria-label]')).toHaveAttribute(
```

Comment:
you can use getByLabelText instead of querySelector. It's more performant than using query selectors: https://testing-library.com/docs/queries/bylabeltext/

Reply:
`getByLabelText` is nice, but in my opinion it makes tests brittle as a change to the text would cause the test to fail. Unless one explicitly cares about the button's text I wouldn't go for it. I personally prefer to have `data-testid` makes life way easier.

Reply:
I disagree. If possible we should select elements by their content as it reflects what the end user sees. If there is no content I would also fall back to using test ids. Imo its more brittle if we select by something like the HTML element. Changing the span to a p would make the test fail even though the element might still look the same to the user.

Reply:
This case especially is interested in seeing specific error messages in specific cases.
If we want to avoid it breaking if we just rephrase an error we could use constants for each individual error and use the same constants in the test.

Reply:
The only downside of `getByLabelText` vs `querySelect` is that the assertion error in case of a wrong expectation is worse:

With query selector:
```
Expected the element to have attribute:
  aria-label="Wrong expectation"
Received:
  aria-label="You need to activate the Safe before transacting"
```

With getByLabelText:
```
TestingLibraryElementError: Unable to find a label with the text of: Wrong expectation

Ignored nodes: comments, script, style
<div>
  <span
    aria-label="You need to activate the Safe before transacting"
    class=""
    data-mui-internal-clone-element="true"
  >
    <button
      disabled=""
    >
      Continue
    </button>
  </span>
</div>
```

If the container is a big html element it becomes really hard to see why the field wasn't found.